mloop/learners.py

```diff
@@ -1626,154 +1840,61 @@ class NeuralNetLearner(Learner, mp.Process):
-                 minimum_uncertainty = 1e-8,
-                 predict_global_minima_at_end = True,
                  **kwargs):
         
         if nn_training_filename is not None:
+            super(NeuralNetLearner,self).__init__(
+                training_filename=nn_training_filename,
+                training_file_type=nn_training_file_type,
+                **kwargs
+            )
+            self.nn_training_file_dir = self.training_file_dir
             
-            nn_training_filename = str(nn_training_filename)
-            # Automatically determine file_type if necessary.
-            if nn_training_file_type is None:
-                nn_training_file_type = mlu.get_file_type(nn_training_filename)
-            nn_training_file_type = str(nn_training_file_type)
-            if not mlu.check_file_type_supported(nn_training_file_type):
-                self.log.error('NN training file type not supported' + repr(nn_training_file_type))
-            self.nn_training_file_dir = os.path.dirname(nn_training_filename)
-            
-            self.training_dict = mlu.get_dict_from_file(nn_training_filename, nn_training_file_type)
-            
-            #Basic optimization settings
-            num_params = int(self.training_dict['num_params'])
-            min_boundary = mlu.safe_cast_to_list(self.training_dict['min_boundary'])
-            max_boundary = mlu.safe_cast_to_list(self.training_dict['max_boundary'])
-            param_names = mlu._param_names_from_file_dict(self.training_dict)
-            
-            #Counters
-            self.costs_count = int(self.training_dict['costs_count'])
-            self.params_count = int(self.training_dict['params_count'])
-            
-            #Data from previous experiment
-            self.all_params = np.array(self.training_dict['all_params'], dtype=float)
-            self.all_costs = mlu.safe_cast_to_array(self.training_dict['all_costs'])
-            self.all_uncers = mlu.safe_cast_to_array(self.training_dict['all_uncers'])
-            
-            self.bad_run_indexs = mlu.safe_cast_to_list(self.training_dict['bad_run_indexs'])            
-            
-            #Derived properties
-            self.best_cost = float(self.training_dict['best_cost'])
-            self.best_params = mlu.safe_cast_to_array(self.training_dict['best_params'])
-            self.best_index = int(self.training_dict['best_index'])
-            self.worst_cost = float(self.training_dict['worst_cost'])
-            self.worst_index = int(self.training_dict['worst_index'])
-            self.cost_range = float(self.training_dict['cost_range'])
-            
-            #Configuration of the fake neural net learner
+            # Configuration of the fake neural net learner
+            # TODO: length_scale and noise_level aren't used by neural net
+            # learner; probably a copy/paste from gaussian process learner.
+            # Maybe they should be deleted?
```

Comment:
:+1:, definitely me copy/pasting (for all of them)

Reply:
Good to know, maybe now's a good time for me to go ahead and delete them then

Reply:
Removed in 423e50d.